Tests/t_ldap_dn.py

```diff
@@ -158,15 +158,42 @@ def test_dn2str(self):
                 [('dc', 'example', 1)],
                 [('dc', 'com', 1)]
             ]),
-            'uid=test\\, 42,ou=Testing,dc=example,dc=com'
+            'uid=test\\2C 42,ou=Testing,dc=example,dc=com'
```

Comment:
API change: I don't find a way to prevent this

Reply:
Even though libldap does some DN normalisation, we can't guarantee to our users that the output of `dn2str` is going to be (character to character) identical. Maybe we should make it explicit that the DN they get from it is always going to be "the same one" but the string representation might evolve without notice? Certainly something that makes them think twice before using it as a dict key, etc.